src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageAccount.cs

```diff
@@ -18,16 +18,19 @@
 using System.Management.Automation;
 using KeyVaultProperties = Microsoft.Azure.Commands.KeyVault.Properties;
 using Microsoft.Azure.Commands.KeyVault.Models.ManagedStorageAccounts;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
 
 namespace Microsoft.Azure.Commands.KeyVault
 {
-    [Cmdlet( VerbsCommon.Get, CmdletNoun.AzureKeyVaultManagedStorageAccount)]
-    [OutputType( typeof( List<PSKeyVaultManagedStorageAccount> ), typeof( PSKeyVaultManagedStorageAccount ), typeof(List<PSDeletedKeyVaultManagedStorageAccountIdentityItem>), typeof(PSDeletedKeyVaultManagedStorageAccount) )]
+    [Cmdlet( VerbsCommon.Get, CmdletNoun.AzureKeyVaultManagedStorageAccount, DefaultParameterSetName = ByAccountNameParameterSet)]
+    [OutputType( typeof( List<PSKeyVaultManagedStorageAccountIdentityItem> ), typeof( PSKeyVaultManagedStorageAccount ), typeof(List<PSDeletedKeyVaultManagedStorageAccountIdentityItem>), typeof(PSDeletedKeyVaultManagedStorageAccount) )]
```

Comment:
we should try to have a single output type, if there is a common ancestor type, and it should be a single type, not an enumeration

Reply:
Re-reading my late night comment, I realize that is the very definition of a base type; I meant that deleted full-entity types are not direct descendants of full-entity types.

Reply:
@dragav It looks like all of the return types are descendants of KVEntityIdentityItem - changing the "OutputType" to be one concrete type will allow users to tab through all the properties of KVEntityIdentityItem when running a cmdlet like "select -Properties", while having multiple output types gives them the option to tab through all properties of all output types, which in this case could be confusing when they are retrieving a regular managedstorageaccount and get the properties for a deleted managedstorageaccount while tabbing.  Note: they can still retreive these properties by typing them out, but they will not be available through tabbing.